#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Make glibc-2.3.5 compile with gcc-4.0 on ia64 fixed by
#		   removing strong_alias.
# DP: Related bugs: 
# DP: Dpatch author: GOTO Masanori <gotom@debian.org>
# DP: Patch author: GOTO Masanori <gotom@debian.org>
# DP: Upstream status: In CVS
# DP: Status Details: 
# DP: Date: 2005-07-16

PATCHLEVEL=0

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
2005-07-16  GOTO Masanori  <gotom@debian.org>

	* sysdeps/ia64/dl-machine.h: Remove strong_alias.

--- sysdeps/ia64/dl-machine.h	2005-07-14 22:21:08.040360346 +0000
+++ sysdeps/ia64/dl-machine.h.new	2005-07-14 22:26:06.513950877 +0000
@@ -263,8 +263,7 @@ elf_machine_runtime_setup (struct link_m
   TRAMPOLINE_TEMPLATE (_dl_runtime_profile, profile_fixup);
 #else
 #define ELF_MACHINE_RUNTIME_TRAMPOLINE				\
-  TRAMPOLINE_TEMPLATE (_dl_runtime_resolve, fixup);		\
-  strong_alias (_dl_runtime_resolve, _dl_runtime_profile);
+  TRAMPOLINE_TEMPLATE (_dl_runtime_resolve, fixup);
 #endif
 
 /* Undo the adds out0 = 16, sp below to get at the value we want in
